homeassistant/components/avreceiver/config_flow.py

```diff
@@ -0,0 +1,128 @@
+            return self.async_abort(reason="single_instance_allowed")
+        await self.async_set_unique_id(f"{DOMAIN}-{serial}")
+        # Show selection form
+        return self.async_show_form(step_id="user")
+
+    async def async_step_user(self, user_input=None):
+        """Get host from user and check connection."""
+        self.hass.data.setdefault(DATA_DISCOVERED_HOSTS, {})
+        # Try connecting to host
+        errors = {}
+        host = None
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            try:
+                avreceiver = await factory(host)
+                await avreceiver.init()
+                await avreceiver.disconnect()
+                self.hass.data.pop(DATA_DISCOVERED_HOSTS)
+                unique_id = (
+                    f"{DOMAIN}-{avreceiver.serial_number or avreceiver.mac or host}"
+                )
+                await self.async_set_unique_id(unique_id, raise_on_progress=False)
+                for entry in self.hass.config_entries.async_entries(DOMAIN):
+                    if entry.data[CONF_HOST] == host:
+                        return self.async_abort(reason="single_instance_allowed")
+                return self.async_create_entry(
+                    title=format_title(host), data={CONF_HOST: host, CONF_ID: unique_id}
+                )
+            except AVReceiverIncompatibleDeviceError:
+                errors[CONF_HOST] = "cannot_connect"
+
+        # Return user input form
+        host_type = (
+            str
+            if not self.hass.data[DATA_DISCOVERED_HOSTS]
+            else vol.In(list(self.hass.data[DATA_DISCOVERED_HOSTS]))
+        )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Required(CONF_HOST, default=host): host_type}),
+            errors=errors,
+        )
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Options for AV Receivers."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Init AV Receiver options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the AV Receiver options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options = {
+            vol.Optional(
+                CONF_ZONE2,
+                default=self.config_entry.options.get(
+                    CONF_ZONE2, DEFAULT_ZONE_DISABLED
+                ),
+            ): bool,
+            vol.Optional(
+                CONF_ZONE3,
+                default=self.config_entry.options.get(
+                    CONF_ZONE3, DEFAULT_ZONE_DISABLED
+                ),
+            ): bool,
+            vol.Optional(
+                CONF_ZONE4,
+                default=self.config_entry.options.get(
+                    CONF_ZONE4, DEFAULT_ZONE_DISABLED
+                ),
+            ): bool,
+        }
```

Comment:
I'd like to make this dynamic to the number of the zones that the receiver actually supports - so it doesn't show Zone3 and Zone4 for my receiver that only has Main and Zone2, for example.